container/client.go

```diff
@@ -0,0 +1,266 @@
+// +build linux
+
+/*
+http://www.apache.org/licenses/LICENSE-2.0.txt
+
+
+Copyright 2017 Intel Corporation
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package container
+
+import (
+	"bufio"
+	"fmt"
+	"os"
+	"path/filepath"
+	"strconv"
+	"strings"
+	"sync"
+	"time"
+
+	log "github.com/Sirupsen/logrus"
+	"github.com/fsouza/go-dockerclient"
+	"github.com/intelsdi-x/snap-plugin-collector-docker/config"
+)
+
+const (
+	dockerVersionKey string = "Version"
+)
+
+// DockerClientInterface provides methods i.a. for interaction with the docker API.
+type DockerClientInterface interface {
+	ListContainersAsMap() (map[string]*ContainerData, error)
+	InspectContainer(string) (*docker.Container, error)
+	FindCgroupMountpoint(string, string) (string, error)
+	FindControllerMountpoint(string, string, string) (string, error)
+	GetDockerParam(string) (string, error)
+}
+
+// DockerClient holds fsouza go-dockerclient instance ready for communication with the server endpoint `unix:///var/run/docker.sock`,
+// cache instance which is used to store output from docker container inspect (to avoid execute inspect request multiply times, it is called only once per container)
+// and diskUsageCollector which is responsible for collecting container disk usage (based on `du -u` command) in the background
```

Comment:
Please check if this comment is up to date - I noticed that `diskUsageCollector` is no longer in DockerClient.